node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -290,14 +315,22 @@ impl<C: Context> ActiveValidator<C> {
             panorama = state.panorama().clone();
         }
         let seq_number = panorama.next_seq_num(state, self.vidx);
+        let prev_round_exp = self
+            .latest_vote(state)
+            .map_or(self.next_round_exp, |vote| vote.round_exp);
+        let next_round_exp = self.round_exp(state, timestamp);
+        if next_round_exp != prev_round_exp {
```

Comment:
Since the round success meter already knows its current round exponent anyway, couldn't it compare these inside `change_exponent`? Then we wouldn't need to access our previous vote and do the comparison here.